google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java

```diff
@@ -286,7 +286,7 @@ public void run() {
                     return;
                   }
 
-                  if (myCurrentSnapshot == null || !myCurrentSnapshot.getId().equals(result.getId())) {
+                  if (isSelectedBeforeTrigger || myCurrentSnapshot == null || !myCurrentSnapshot.getId().equals(result.getId())) {
```

Comment:
@elharo we could do that. That would mean though that if you already have a snapshot selected, and you click on it again it would do a redundant reloading of the result.